seoulai_gym/envs/traders/market.py

```diff
@@ -0,0 +1,221 @@
+"""
+Stock Market
+Cinyoung Hur, cinyoung.hur@gmail.com
+James Park, laplacian.k@gmail.com
+seoulai.com
+2018
+"""
+import copy
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+import pygame
+from pygame.locals import QUIT
+
+from seoulai_gym.envs.traders.base import Constants
+from seoulai_gym.envs.traders.price import Price
+from seoulai_gym.envs.traders.graphics import Graphics
+# from seoulai_gym.envs.traders.rules import Rules
+
+
+class Market():
+    def __init__(
+        self,
+        state: str=None,
+    ) -> None:
+        """Initialize market and its visualization.
+        Args:
+            state: Optional, path to saved game state. TODO
+        Returns:
+            None
+        """
+        self.init()
+
+        # graphics is for visualization
+        self.graphics = Graphics()
+
+    def init(
+        self,
+    ):
+        self.price = Price()    # TODO: data generator
+        self.tick = 0
+        self.max_tick_size = 1000
+
+    def select(
+        self,
+        exchang_name: str,
+    ):
+
+        # TODO : add some exchanges. ex. bithumb, bittrex, coinone, binance...
+        # TODO : fixed parameters(fee ratio...) can't be edited.
+        if exchang_name == "upbit":
+            self.fee_rt = 0.05/100
+        else:
+            self.fee_rt = 0.10/100
+
+    def reset(
+        self
+    ) -> List:
+        """Reset all variables and initialize new game.
+        Returns:
+            obs: Information about trading parameters.
+        """
+
+        self.init()
+
+        obs = [self.price.price_list[:1], self.fee_rt]
+        return obs
+
+    def step(
+        self,
+        agent,
+        decision,
+        trad_price: float,
+        trad_qty: int,
+    ):
+        """Make a step (= move) within market.
+        Args:
+            agent: Agent making a move.
+            from_row: Row of board of original piece location.
```

Comment:
This is comment for checkers.

Reply:
I just fixed and push it.

Reply:
@laplacian-k where did you push?

Reply:
Sorry, I pushed in laplacian-k/gym yesterday.
And I just git clone trader branch. then, I tried to push fixed market.py file in seoulai/gym, trader branch. but I couldn't push that file

> git push origin trader
> remote: Permission to seoulai/gym.git denied to laplacian-k.
> ...
How can I do?

Reply:
@laplacian-k let me check it

Reply:
@laplacian-k 
Try now!